tests/platform_tests/api/test_fan.py

```diff
@@ -41,19 +41,63 @@
 class TestFanApi(PlatformApiTestBase):
 
     num_fans = None
+    fan_truth = None
+    duthost_vars = None
 
     # This fixture would probably be better scoped at the class level, but
     # it relies on the platform_api_conn fixture, which is scoped at the function
     # level, so we must do the same here to prevent a scope mismatch.
 
     @pytest.fixture(scope="function", autouse=True)
-    def setup(self, platform_api_conn):
+    def setup(self, duthost, platform_api_conn):
         if self.num_fans is None:
             try:
                 self.num_fans = int(chassis.get_num_fans(platform_api_conn))
             except:
                 pytest.fail("num_fans is not an integer")
 
+        # Get fan truths from platform.json file
+        chassis_truth = duthost.facts.get("chassis")
+        if chassis_truth:
+            self.fan_truth = chassis_truth.get('fans', None)
+            if not self.fan_truth:
+                logger.warning("Unable to get fan_truth from platform.json, test results will not be comprehensive")
+        else:
+            logger.warning("Unable to get chassis_truth from platform.json, test results will not be comprehensive")
+
+        # Get host vars from inventory file
+        self.duthost_vars = duthost.host.options['inventory_manager'].get_host(duthost.hostname).vars
```

Comment:
This data only needs to be collected once for the class, not for every function. Can you move this logic into a class-scoped fixture like I did here: https://github.com/Azure/sonic-mgmt/pull/2209/files#diff-5d04a73b4b142807edb8c343db1d80dbR55?

Reply:
fixed